src/components/canvas-tools/drawing-tool/toolbar.tsx

```diff
@@ -0,0 +1,289 @@
+// tslint:disable:jsx-no-lambda
+import * as React from "react";
+import { DrawingContentModelType, Color, ToolbarModalButton, TOOLBAR_WIDTH,
+  colors, computeStrokeDashArray } from "../../../models/tools/drawing/drawing-content";
+import { ToolTileModelType } from "../../../models/tools/tool-tile";
+import { observer } from "mobx-react";
+
+export interface TextButtonData {
+  color: string;
+}
+
+export interface PolygonButtonData {
+  type: string;
+  stroke?: string;
+  fill?: string;
+}
+
+export interface LineButtonData {
+  lineColor: Color;
+}
+
+export interface ToolbarFlyoutViewProps {
+  selected: boolean;
+}
+
+export interface ToolbarFlyoutViewState {
+  open: boolean;
+  selectedIndex: number;
+}
+
+export interface FlyoutMenuItem {
+  title: string;
+  icon: string;
+}
+
+export class ToolbarFlyoutView extends React.Component<ToolbarFlyoutViewProps, ToolbarFlyoutViewState> {
+  constructor(props: ToolbarFlyoutViewProps){
+    super(props);
+
+    this.state = {
+      open: false,
+      selectedIndex: 0
+    };
+  }
+
+  public render() {
+    const {open, selectedIndex} = this.state;
+    const children = this.props.children as any;
+    const selected = children ? children[selectedIndex] : null;
+    if (!selected) {
+      return null;
+    }
+    const {props} = selected;
+    const className = this.props.selected ? "button selected" : "button";
+    return (
+      <div className="flyout-top-button">
+        <div className={className} title={props.title} style={props.style}
+            onClick={(e) => this.handleChildClick(e, selectedIndex)}>
+          {props.children}
+        </div>
+        {open ? this.renderOpen() : null}
+        <div className="flyout-toggle" onClick={this.handleToggleOpen}>{open ? "▼" : "▶"}</div>
+      </div>
+    );
+  }
+
+  private handleToggleOpen = () => {
+    this.setState({open: !this.state.open});
+  }
+
+  private handleChildClick = (e: React.MouseEvent<HTMLDivElement>, index: number) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({selectedIndex: index, open: false});
+    const children = this.props.children as any;
+    const selected = children ? children[index] : null;
```

Comment:
This had been safe, because `children` were being set explicitly, but then I realized I refactored and made this unused, and forgot to blow it all away. Fixed.